src/usr/lib/alternc/generate_certbot.php

```diff
@@ -91,7 +104,50 @@ function vprint( $message, $params ){
     vprint(_("Skipping system certificates, requested certificates type: %s\n"), array($REQUEST_CERTS));
 }
 
-if ($REQUEST_CERTS == 'all' || $REQUEST_CERTS == 'non-system') {
+if ($REQUEST_CERTS == 'specific' && $domains !== NULL) {
+    foreach ($domains as $domain) {
+        if (!$certbot->isLocalAlterncDomain($domain)) {
+            vprint(_("Skipping '%s', the domain is not a local AlternC domain"),
+                   array($domain));
+            continue;
+        }
+        foreach ($certs = $ssl->get_valid_certs($domain, 'letsencrypt') as $index => $certificate) {
+            // Remove the wildcard
+            if ($certificate['id'] == 0) {
+                unset($certs[$index]);
+            }
+
+            // Remove those whose providers do not match
+            if ($certificate['provider'] != 'letsencrypt') {
+                unset($certs[$index]);
+            }
+
+            // Unless this domain is the default fqdn, remove it.
+            if ($domain != $ssl->default_certificate_fqdn &&
+                $certificate['fqdn'] == $ssl->default_certificate_fqdn) {
+                unset($certs[$index]);
+            }
+        }
+
+        // Request if there are no valid certificates other than the defaults removed,
+        // or if the request is forced.
+        if (sizeof($certs) <= 0  || $force_request) {
+            vprint(_("Requesting import of certificate for '%s'"), array($domain));
+            $sub_domain = m_dom::get_sub_domain_id_and_member_by_name($domain);
+            if ($sub_domain === FALSE) {
+                vprint(_("Failed to find sub_domain id and member by name for '%s'"), array($domain));
+                continue;
+            }
+            $mem->su($sub_domain['member_id']);
+            $certbot->import($domain);
```

Comment:
add force option 
following #74

Reply:
Hello

#74 is backported to master; could you update your code to integrate force mode ?